test/iifInterfaceTests.jl

```diff
@@ -0,0 +1,437 @@
+        :max => VariableEstimate(:default, :max, [10.0]),
+        :mean => VariableEstimate(:default, :mean, [5.0]),
+        :modefit => VariableEstimate(:default, :modefit, [7.0]))
+
+    # Confirm they're different
+    @test estimates(newvar) != estimates(var)
+    # Persist it.
+    updateVariableSolverData!(dfg, newvar)
+    # Get the latest
+    var = getVariable(dfg, :a)
+    @test symdiff(collect(keys(estimates(var))), [:default, :second]) == Symbol[]
+
+    #Check if variable is updated
+    @test estimates(newvar) == estimates(var)
+
+
+    # Delete :default and replace to see if new ones can be added
+    delete!(estimates(newvar), :default)
+    #confirm delete
+    @test symdiff(collect(keys(estimates(newvar))), [:second]) == Symbol[]
+    # Persist it.
+    updateVariableSolverData!(dfg, newvar)
+
+    # Get the latest and confirm they're the same, :second
+    var = getVariable(dfg, :a)
+
+    # TODO issue #166
+    @test estimates(newvar) != estimates(var)
+    @test collect(keys(estimates(var))) ==  [:default, :second]
+
+    # @test symdiff(collect(keys(estimates(getVariable(dfg, :a)))), [:default, :second]) == Symbol[]
+end
+
+# Connectivity test
+@testset "Connectivity Test" begin
+    global dfg,v1,v2,f1
+    @test isFullyConnected(dfg) == true
+    @test hasOrphans(dfg) == false
+    addVariable!(dfg, :orphan, ContinuousScalar, labels = [:POSE])
+    @test isFullyConnected(dfg) == false
+    @test hasOrphans(dfg) == true
+end
+
+# Adjacency matrices
+@testset "Adjacency Matrices" begin
+    global dfg,v1,v2,f1
+    # Normal
+    adjMat = getAdjacencyMatrix(dfg)
+    @test size(adjMat) == (2,4)
+    @test symdiff(adjMat[1, :], [nothing, :a, :b, :orphan]) == Symbol[]
+    @test symdiff(adjMat[2, :], [:abf1, :abf1, :abf1, nothing]) == Symbol[]
+    #sparse
+    adjMat, v_ll, f_ll = getAdjacencyMatrixSparse(dfg)
+    @test size(adjMat) == (1,3)
+
+    # Checking the elements of adjacency, its not sorted so need indexing function
+    indexOf = (arr, el1) -> findfirst(el2->el2==el1, arr)
+    @test adjMat[1, indexOf(v_ll, :orphan)] == 0
+    @test adjMat[1, indexOf(v_ll, :a)] == 1
+    @test adjMat[1, indexOf(v_ll, :b)] == 1
+    @test symdiff(v_ll, [:a, :b, :orphan]) == Symbol[]
+    @test symdiff(f_ll, [:abf1, :abf1, :abf1]) == Symbol[]
+end
+
+# Deletions
+@testset "Deletions" begin
+    deleteFactor!(dfg, :abf1)
+    @test getFactorIds(dfg) == []
+    deleteVariable!(dfg, :b)
+    @test symdiff([:a, :orphan], getVariableIds(dfg)) == []
+    #delete last also for the LightGraphs implementation coverage
+    deleteVariable!(dfg, :orphan)
+    @test symdiff([:a], getVariableIds(dfg)) == []
+    deleteVariable!(dfg, :a)
+    @test getVariableIds(dfg) == []
+end
+
+
+# Now make a complex graph for connectivity tests
+numNodes = 10
+#the deletions in last test should have cleared out the fg
+# dfg = DistributedFactorGraphs._getDuplicatedEmptyDFG(dfg)
+# if typeof(dfg) <: CloudGraphsDFG
+#     clearSession!!(dfg)
+# end
```

Comment:
I serves as additional tests not to clear the session

Reply:
good point, I frequency use `deepcopy(fg)` all over the place.  A way to clone or duplicate factor graphs will be pretty important i think